packages/plugin-odoo-snippets/src/OdooSnippet.ts

```diff
@@ -297,6 +317,30 @@ export class OdooSnippet<T extends JWPluginConfig = JWPluginConfig> extends JWPl
         const cover = covers && covers[0];
         if (cover) return domLayout.getDomNodes(cover)[0];
     }
+    async getSelectedText(params: CommandParams): Promise<string> {
+        const selectedNode = params.context.range.selectedNodes().map(node => node.clone());
+        const container = new ContainerNode();
+        container.append(...selectedNode);
+        const renderer = this.editor.plugins.get(Renderer);
+        const renderedNode = await renderer.render<HTMLElement[]>(
+            HtmlDomRenderingEngine.id,
+            container,
+        );
+        // todo: what if there is more rendered nodes?
+        return renderedNode ? renderedNode[0].innerText : '';
+    }
+    async getSelectedLink(params: CommandParams): Promise<string> {
+        const selectedNode = params.context.range.selectedNodes().map(node => node.clone());
+        const nodeWithFormat: InlineNode = selectedNode.find(
+            node =>
+                node instanceof InlineNode &&
+                node.formats.find(format => format instanceof LinkFormat),
+        ) as InlineNode;
+        const linkFormat: LinkFormat =
+            nodeWithFormat &&
+            (nodeWithFormat.formats.find(format => format instanceof LinkFormat) as LinkFormat);
+        return linkFormat ? linkFormat.url : '';
+    }
```

Comment:
All this code is irrelevant now that modifiers were merged. Also there is already a function doing exactly this in `Link`.